books/epubinfo.py

```diff
@@ -134,18 +134,59 @@ def _get_subject(self):
         return subjectlist
 
     def _get_cover_image(self):
-        # TODO check if cover is image
+        cover = None
         meta_content_cover = None
         for element in self._e_metadata.iterfind(
                 '{http://www.idpf.org/2007/opf}meta'):
             if element.get('name') == 'cover':
                 meta_content_cover = element.get('content')
 
+        # ePub 2 metadata
         if meta_content_cover:
             for node in self._tree.iter():
                 if node.tag == '{http://www.idpf.org/2007/opf}item':
                     if node.attrib['id'] == meta_content_cover:
                         cover = node.attrib['href']
+            # In case meta tag refers to cover itself
+            if cover is None:
+                cover = meta_content_cover
+
+        # Guide
+        if cover is None:
+            for node in self._tree.iter():
+                if node.tag == '{http://www.idpf.org/2007/opf}reference':
```

Comment:
Since all the next comparisons follow the same structure, can you merge them into one to avoid traversing the time a bunch of times, ie:

```python
if cover is None:
    for node in self._tree.iter():
        # epub foo
        if node.tag = ...... :
            if node.attrib['type'] == 'cover':
                cover = ....
                break
        # epub bar
        elif node.tag = ....
```
The break is basically for exiting the `for` when a match is found (not sure if any format/spec allows several elements of the same cover type, but just in case).